pytorchfi/core.py

```diff
@@ -44,7 +45,7 @@ def __init__(self, model, batch_size, input_shape=None, layer_types=None, **kwar
         if len(layer_types) < 0:
             raise AssertionError("Error: At least one layer type must be selected.")
 
-        handles, _shapes = self._traverse_model_set_hooks(
+        handles, _shapes, self.weights_size = self._traverse_model_set_hooks(
             self.original_model, self._inj_layer_types
```

Comment:
Given that `self._traverse_model_set_hooks` is an internal function, do we need to pass the two class variables as parameters? We can

1. read it directly from inside the function (probably what we should do)
2. make the function available via api and generalize parameters

This is also out of scope for this PR so we can change in a follow up PR

Reply:
Good point! Let's address this in follow-up PR